optimum/habana/transformers/models/qwen2/modeling_qwen2.py

```diff
@@ -899,11 +934,15 @@ def prepare_inputs_for_generation(
 
 def apply_customized_rope(q, k, cos, sin, position_ids):
     if q.device.type == "hpu" and FusedRoPE:
-        # TODO: remove `.clone()` when it is fixed in SynapseAI
-        return FusedRoPE.apply(
-            q, cos.unsqueeze(0).unsqueeze(0).clone(), sin.unsqueeze(0).unsqueeze(0).clone(), position_ids
-        ), FusedRoPE.apply(
-            k, cos.unsqueeze(0).unsqueeze(0).clone(), sin.unsqueeze(0).unsqueeze(0).clone(), position_ids
-        )
+       # TODO: remove `.clone()` when it is fixed in SynapseAI
+        if k.dtype == torch.bfloat16:
```

Comment:
when this statement is ```False``` entire function ends up without returning value. I guess it was assumed to fallback to default ```apply_rotary_pos_emb``` in case of non-bf16? If so, current if-statement should be merged with upper one

Reply:
thanks, fixed